instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/crash/NdkCrashHandlerE2ETests.kt

```diff
@@ -0,0 +1,113 @@
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import com.datadog.android.nightly.TEST_METHOD_NAME_KEY
+import com.datadog.android.nightly.rules.NightlyTestRule
+import com.datadog.android.nightly.services.NdkCrashService
+import com.datadog.android.nightly.services.NdkHandlerDisabledNdkCrashService
+import com.datadog.android.nightly.services.RumEnabledNdkCrashService
+import com.datadog.android.nightly.utils.initializeSdk
+import com.datadog.android.nightly.utils.stopSdk
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class NdkCrashHandlerE2ETests {
+
+    // region Tests
+
+    @get:Rule
+    val nightlyTestRule = NightlyTestRule()
+
+    /**
+     * apiMethodSignature: com.datadog.android.core.configuration.Configuration$Builder#fun build(): Configuration
+     * apiMethodSignature: com.datadog.android.core.configuration.Configuration$Builder#fun addPlugin(com.datadog.android.plugin.DatadogPlugin, com.datadog.android.plugin.Feature): Builder
+     */
+    @Test
+    fun crash_reports_rum_enabled() {
+        // We initialize the SDK in this process as both processes are sharing the same data
+        // storage space. Flushing the data in this process at the end of this test will also
+        // flush the data produced by the service process. Besides this our SDK has a safety
+        // measure to prevent sending an even twice from 2 different process. For this reason
+        // we are using a NoOpOkHttpUploader in case the process is not the app main process.
+        val testMethodName = "ndk_crash_reports_rum_enabled"
+        initializeSdk(InstrumentationRegistry.getInstrumentation().targetContext)
+        startService(
+            testMethodName,
+            RumEnabledNdkCrashService::class.java
+        )
+        waitForProcessToIdle()
+        stopService(RumEnabledNdkCrashService::class.java)
+        // we stop and initialize the SDK again to handle the preserved ndk crash
+        stopSdk()
+        initializeSdk(InstrumentationRegistry.getInstrumentation().targetContext)
+        waitForProcessToIdle()
+    }
+
+    /**
+     * apiMethodSignature: com.datadog.android.core.configuration.Configuration$Builder#fun build(): Configuration
+     * apiMethodSignature: com.datadog.android.core.configuration.Configuration$Builder#fun addPlugin(com.datadog.android.plugin.DatadogPlugin, com.datadog.android.plugin.Feature): Builder
+     */
+    @Test
+    fun crash_reports_config_disabled() {
+        val testMethodName = "ndk_crash_reports_feature_disabled"
+        initializeSdk(InstrumentationRegistry.getInstrumentation().targetContext)
+        startService(
+            testMethodName,
+            NdkHandlerDisabledNdkCrashService::class.java
+        )
+        waitForProcessToIdle()
+        stopService(NdkHandlerDisabledNdkCrashService::class.java)
```

Comment:
why we don't restart SDK as above to make sure that nothing is sent?

Reply:
Ahh my bad, yes we should do this also here. Tks !!